### ISSUE TITLE
Places: calling placesNearby with the near_page_token of a previous placesNearby gives an error

### ISSUE THREAD

**Issue by ozomer** [googlemaps/google-maps-services-js#139]
I tried building a function that will keep calling placesNearby until all pages are fetched (each page contains up to 20 reports).

```
const googleMapsClient = require('@google/maps').createClient({ key: "SECRET" });

function allPlacesNearby(query, soFar = []) {
  console.info(`sending placesNearby query: ${JSON.stringify(query)}`);
  return new Promise((resolve, reject) => googleMapsClient.placesNearby(query, (err, response) => (err ? reject(err) : resolve(response))))
  .then(result => {
    console.info(`received ${result.json.results.length} results`);
    const allResults = soFar.concat(result.json.results);
    const { next_page_token: nextPageToken } = result.json;
    if (nextPageToken) {
      return allPlacesNearby({
        "location": query.location,
        "pagetoken": nextPageToken
      }, allResults);
    }
    return allResults;
  });
}
```

The following call fails (prints the error):
```
allPlacesNearby({"location":[32.05091874118154,34.77247240956386],"radius":1800,"type":"cafe"}).then(() => console.log('success')).catch(err => console.error(`err: ${err}`))
```

However, if I add a 5 seconds delay between calls like:
```
function allPlacesNearby(query, soFar = []) {
  console.info(`sending placesNearby query: ${JSON.stringify(query)}`);
  return new Promise((resolve, reject) => googleMapsClient.placesNearby(query, (err, response) => (err ? reject(err) : resolve(response))))
  .then(result => {
    console.info(`received ${result.json.results.length} results`);
    const allResults = soFar.concat(result.json.results);
    const { next_page_token: nextPageToken } = result.json;
    if (nextPageToken) {
      return new Promise(resolve => setTimeout(resolve, 5000)).then(() => allPlacesNearby({
        "location": query.location,
        "pagetoken": nextPageToken
      }, allResults));
    }
    return allResults;
  });
}
```
The operation will succeed!

Am I missing something?

P.S. getting all the pages results should be in the lib itself in my opinion.

**Comment by ozomer** [comment]
Apparently this is not a problem with the nodejs package, but a problem with the api. I implemented the same code with simple http requests and the result was the same - a ~1 second time gap is required between each query request.

I'm leaving this issue open because I still think this is something that should be implemented in the nodejs package.

**Comment by Jip-Hop** [comment]
I think I'm running into this issue as well while adapting the example code from [the tests](https://github.com/googlemaps/google-maps-services-js/blob/master/spec/e2e/places-spec.js).

```var Promise = require('q').Promise;
var googleMaps = require('@google/maps').createClient({
    key: 'key',
    Promise: require('q').Promise
});

function fetchPlaces(location, radius, done, fail){
    googleMaps.placesNearby({
        location: location,
        radius: radius
    })
    .asPromise()
    .then(function(response) {
        console.log(response.json);
        //expect(response.json.next_page_token).not.toBeFalsy();
        function getNextPage() {
        return googleMaps.placesNearby({
            pagetoken: response.json.next_page_token
        }).asPromise();
        }
        return getNextPage()
            .then(function repeatWhileInvalid(nextResponse) {
            if (nextResponse.json.status !== 'INVALID_REQUEST') {
                return nextResponse;
            }

            // Wait one second, and try again.
            return new Promise(function(resolve) {
                setTimeout(resolve, 1000);
            })
            .then(getNextPage)
            .then(repeatWhileInvalid);
            });
    })
    .then(function(nextResponse) {
        //expect(nextResponse.json.status).toBe('OK');
        //expect(nextResponse.json.results.length).not.toBeFalsy();
    })
    .then(done, fail);
}

fetchPlaces([-33.86746, 151.207090], 5000, function(){
    console.log("Done")
}, function(e){
    console.log("Fail")
    console.log(e);
});
```

The response is `InvalidValueError: missing property "location"`.

**Comment by ozomer** [comment]
@Jip-Hop This is a different error which I also encountered. The library requires the `location` property to be in every call, even though the HTTP api does not require the `location` property when `pagetoken` is set.
From what I recall there is no effect to also pass the `location` when asking for the following pages. However, a 1-2 seconds delay is indeed necessary for the `pagetoken` to become valid and be accepted.

Personally I just switched to the HTTP api directly using [request-promise-native](https://www.npmjs.com/package/request-promise-native).
